### PR TITLE
cmd/snap: warn when a snap is not from the tracking channel

### PR DESCRIPTION
**Pull request by chipaca**
`snap revert` and `snap refresh --revision` can leave a snap with its
current revision not coming from the channel the snap is tracking. As
this can be surprising to some users, print a little message abou tit.

### CODE REVIEW

File: cmd/snap/cmd_snap_op.go [hunk 1]
```diff
@@ -357,20 +357,34 @@ func showDone(names []string, op string) error {
...
 			}
+		case "revert":
+			// TRANSLATORS: first %s is a snap name, second %s is a revision
+			fmt.Fprintf(Stdout, i18n.G("%s reverted to %s\n"), snap.Name, snap.Version)
 		default:
 			fmt.Fprintf(Stdout, "internal error, unknown op %q", op)
 		}
+		if snap.TrackingChannel != snap.Channel {
+			// TRANSLATORS: first %s is a snap name, following two %s are channel names
+			fmt.Fprintf(Stdout, i18n.G("This leaves %s tracking %s, but the current revision is from %s.\n"),
```
Comment (niemeyer) [inline cmd/snap/cmd_snap_op.go]: I have strong suspicions that this isn't going to work so well, and will end up confusing people. The problem is that snaps do not have an intrinsic channel. Something that is in candidate today can be in stable tomorrow, and we won't know the difference. The Channel field in SideInfo comes from a snapshot of when we observed the snap, but we don't know what happens next.

Comment (chipaca) [reply cmd/snap/cmd_snap_op.go]: OK, so how do we fix it? Users are getting confused by the behaviour _today_.

Comment (chipaca) [reply cmd/snap/cmd_snap_op.go]: (from the standup), i'll be dropping the mention of the channel "from", and the switch suggestion, the rest changes ok

File: cmd/snap/cmd_snap_op.go [hunk 2]
```diff
@@ -357,20 +357,32 @@ func showDone(names []string, op string) error {
...
 			}
+		case "revert":
+			// TRANSLATORS: first %s is a snap name, second %s is a revision
+			fmt.Fprintf(Stdout, i18n.G("%s reverted to %s\n"), snap.Name, snap.Version)
 		default:
 			fmt.Fprintf(Stdout, "internal error, unknown op %q", op)
 		}
+		if snap.TrackingChannel != snap.Channel {
+			// TRANSLATORS: first %s is a snap name, following %s is a channel name
+			fmt.Fprintf(Stdout, i18n.G("This leaves %s tracking %s.\n"), snap.Name, snap.TrackingChannel)
```
Comment (mvo5) [inline cmd/snap/cmd_snap_op.go]: maybe: "%s is tracking %s (check `snap switch` to switch channels)" ? But I know this is bike-sheding and its a good question if the mention of switch is helpful or just adds noise.

**Pull request merged**